src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -105,23 +107,31 @@ private WindowsIdentity()
         /// <summary>
         /// Initializes a new instance of the WindowsIdentity class for the user represented by the specified User Principal Name (UPN).
         /// </summary>
-        /// <remarks>
-        /// Unlike the desktop version, we connect to Lsa only as an untrusted caller. We do not attempt to exploit Tcb privilege or adjust the current
-        /// thread privilege to include Tcb.
-        /// </remarks>
-        public WindowsIdentity(string sUserPrincipalName)
+        public WindowsIdentity(string sUserPrincipalName, string type)
             : base(null, null, null, ClaimTypes.Name, ClaimTypes.GroupSid)
+        {
+            KerbS4ULogon(sUserPrincipalName, ref _safeTokenHandle);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the WindowsIdentity class for the user represented by the specified User Principal Name (UPN).
+        /// </summary>
+        public WindowsIdentity(string sUserPrincipalName)
+            : this(sUserPrincipalName, null)
+        { }
+
+        private static void KerbS4ULogon(string sUserPrincipalName, ref SafeAccessTokenHandle accessTokenHandle)
         {
             // Desktop compat: See comments below for why we don't validate sUserPrincipalName.
 
-            using (SafeLsaHandle lsaHandle = ConnectToLsa())
+            // 8 byte or less name that indicates the source of the access token. This choice of name is visible to callers through the native GetTokenInformation() api
+            // so we'll use the same name the CLR used even though we're not actually the "CLR."
+            byte[] sourceName = { (byte)'C', (byte)'L', (byte)'R', (byte)0 };
```

Comment:
Can we cache this array as a static readonly byte[]?

Reply:
I'll do it